src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -144,7 +147,13 @@ def load_command_table(self, command_loader):
                 _parser=command_parser)
 
     def validation_error(self, message):
-        az_error = ValidationError(message)
+        # Get command to provide to error help
+        full_command = self.prog + ' ' + self.full_raw_command
```

Comment:
What is the function of `self.prog`?

Reply:
This is the name and action of the resource such as "batch create". Together with the other part we need to get the whole command line. I refactor the code and hope that will be clearer.